logic.js

```diff
@@ -0,0 +1,176 @@
+            const getcommentId = ()=> commentId
+            const setcommentId = (value) => {
+                commentId= value
+            }
+            return{
+                getcommentContent,
+                setcommentContent,
+                getId: getcommentId,
+                setId: setcommentId
+            }
+        }
+        // =====================End of comment module==================================
+        const getTweetContent = () => tweetText
+        const setTweetContent = (value) => {
+            tweetText = value;
+        }
+
+        const getTweetId = () => tweetId
+        const setTweetId = (value) => {
+            tweetId = value;
+        }
+
+        // add a comment to the tweet
+        const addComment = function(value) {
+            const comment  = Comment(value)
+            comment.setId("T"+getTweetId()+"-C"+comments.length)
+            addElement(comments, comment)
+            return comment
+        } 
+        // remove comment from the tweet
+        const removeCommentById = (id) => {
+            comments = removeElementById(comments, id)
+        }
+
+        // get specific comment by its id/index
+        const getCommentById = (id) => {
+            return getElementById(comments, id)
+        }
+
+        const getNumberOfComments = () => {
+            return comments.length
+        }
+
+        const getComments = () => {
+            return comments
+        }
+
+        return{
+            getTweetContent,
+            setTweetContent,
+            getId: getTweetId,
+            setId : setTweetId,
+            addComment,
+            removeCommentById,
+            getCommentById,
+            getNumberOfComments,
+            getComments
+        }
+
+    }
+    // =======================End of tweet module=========================================
+
+    // Post a tweet
+    const postTweet = (tweetContent) => {
+        const tweet = Tweet(tweetContent)
+        tweet.setId(allTweets.length)
+        addElement(allTweets, tweet)
+        return tweet
+    }
+
+    // get specific tweet by its id/index
+    const getTweetById = (id) => {
+        return getElementById(allTweets, id)
+    }
+
+    // delete a specific tweet based on its id/index
+    const removeTweetById = (id) => {
+        // allTweets array is being called as value, and not as reference, that's why it has been reassigned again
+        allTweets = removeElementById(allTweets, id)
+        if(allTweets.length == 0){
+            console.log(' there are no posts ')
+        }
+        for(let i=0; i<allTweets.length; i++){
+            console.log('Tweet of id: '+ allTweets[i].getId()+' exists')
+        }
+    }
+
+    const getAllTweets = () => {
+        return allTweets
+    }
+
+    return {
+        postTweet,
+        getTweetById,
+        removeTweetById,
+        getPosts: getAllTweets
+    }
+
+}
+// ================End of Tweets module================================================
```

Comment:
try to simplify your code, instead of adding a lot of boilerplate, function etc.